src/core/test/spec.js

```diff
@@ -87,81 +86,124 @@ describe('Test Passables "test" function', () => {
-                        expect(output.hasErrors(f3)).to.equal(false);
+                        expect(output.hasErrors(f5)).to.equal(false);
                         done();
                     }, 500);
                 });
+
             });
 
             describe('When passing a Promise as a test', () => {
                 describe('failing', () => {
-                    let f1, f2;
+                    let f1, f2, f3, f4;
+                    const rejectionMessage = lorem.sentence();
+
                     beforeEach(() => {
                         f1 = lorem.word();
                         f2 = lorem.word();
+                        f3 = lorem.word();
+                        f4 = lorem.word();
 
-                        output = passable(lorem.word(), (test, draft) => {
-                            test(f1, lorem.sentence(), new Promise((resolve, reject) => setImmediate(reject)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve, 500)));
+                        output = passable(lorem.word(), (test) => {
+                            test(f1, f1, new Promise((_, reject) => setImmediate(reject)));
+                            test(f2, new Promise((_, reject) => setImmediate(() => reject(rejectionMessage))));
+                            test(f3, lorem.sentence(), new Promise((_, reject) => setImmediate(() => reject(lorem.word()))));
+                            test(f3, f3, new Promise((_, reject) => setImmediate(() => reject(rejectionMessage))));
+                            test(f4, f4, new Promise((resolve) => setTimeout(resolve)));
+                            test(f4, f4, new Promise((resolve) => setTimeout(resolve, 500)));
                             test(lorem.word(), lorem.sentence(), noop);
                         });
                     });
 
                     it('Should immediately register tests', () => {
-                        expect(output.testCount).to.equal(4);
+                        expect(output.testCount).to.equal(7);
                     });
 
                     it('Should run async test promise', (done) => {
                         passable(lorem.word(), (test) => {
-                            test(f1, lorem.sentence(), new Promise((resolve) => done()));
+                            test(f1, lorem.sentence(), new Promise(() => done()));
                             test(lorem.word(), lorem.sentence(), noop);
                         });
                     });
 
+                    it('Should fail with rejection message when provided', (done) => {
+                        setTimeout(() => {
+                            [f2, f3].forEach((field) => {
+                                expect(output.getErrors(field)).to.include(rejectionMessage);
+                                expect(output.getErrors(field)).not.to.include(field);
```

Comment:
What does the second line test exactly?

Reply:
It's to check that the test statement is not part of the errors. I can remove it if you would prefer